app/src/main/java/com/x8bit/bitwarden/ui/platform/components/image/BitwardenGifImage.kt

```diff
@@ -0,0 +1,51 @@
+package com.x8bit.bitwarden.ui.platform.components.image
+
+import android.content.res.Configuration
+import androidx.annotation.DrawableRes
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalConfiguration
+import com.bumptech.glide.integration.compose.ExperimentalGlideComposeApi
+import com.bumptech.glide.integration.compose.GlideImage
+
+/**
+ * A composable that displays a gif image.
+ *
+ * The content will also be scaled to fit the image to the container.
+ *
+ * **NOTE:** Due to using Glide under the hood, this does not respect the themed resource management
+ * of the passed in [resId] automatically. If there is a specific version of the `.gif` you can pass
+ * in the optional [darkModeResId]. You would need to save these in your resources with unique
+ * names.
+ *
+ * @sample
+ * ```Kotlin
+ * BitwardenGifImage(
+ *   resId = R.drawable.img_setup_autofill_light,
+ *   darkModeResId = R.drawable.img_setup_autofill_dark,
+ * )
+ * ```
+ */
+@OptIn(ExperimentalGlideComposeApi::class)
+@Composable
+fun BitwardenGifImage(
+    @DrawableRes resId: Int,
+    modifier: Modifier = Modifier,
+    @DrawableRes darkModeResId: Int? = null,
+    contentDescription: String? = null,
+) {
+    val uiMode = LocalConfiguration.current.uiMode and Configuration.UI_MODE_NIGHT_MASK
+    GlideImage(
```

Comment:
After making this change:

https://github.com/user-attachments/assets/e4bd15a6-a183-4c35-b205-b45155ce5e22

Reply:
Can we not just put the dark mode version in the dark mode resource folder?

Reply:
Maybe I need to re-phrase the comment, that is what gives this result.

Reply:
🤔

Reply:
Weird. 

For posterity, it looks like it's tracked [here](https://github.com/bumptech/glide/issues/3751). It's clearly not fixed in the latest release. 😢 

There's a workaround [here](https://github.com/bumptech/glide/issues/3778#issuecomment-517909978) that seems to work for people. I'm assuming it will work in Compose since the only different is setting `model` to a Drawable instead of a resource ID.

Reply:
Did a quick check and I'm getting the expected results by passing a Drawable instead of the resource ID. 

Just so it's said, someone in one of the Glide threads did mention encountering "more OOM exceptions" after making this change, but I can't imagine one small gif causing problems. 🤷 

FYI; I used the emulators Device UI shortcuts to change between themes in the video, but it works the same when using the tile.

https://github.com/user-attachments/assets/d8bf53e0-7220-41ef-bd84-6975739087d2

Reply:
Hmmm, I agree that a single gif shouldn't be an issue.
Ideally I think we would use some thing like [this](https://github.com/google/accompanist/blob/main/drawablepainter/src/main/java/com/google/accompanist/drawablepainter/DrawablePainter.kt) but we should be fine as is.

Maybe check on this as we update glide in the future.